src/reducer.js

```diff
@@ -13,38 +13,71 @@ function startHere(state: {} = {}) {
   return state;
 }
 
+function addTodo(
+  state: Array<{
+    uid: string,
+    target: {},
+    actionId: string,
+    completed: boolean}>,
+  actionId: string,
+  phoneNumber: {}) {
+  // If there are already todos for this action, do nothing
+  if (_.find(state, { actionId, target: phoneNumber })) {
+    return state;
+  }
+  return state.concat(
+    {
+      actionId,
+      target: phoneNumber,
+      uid: uuid.v4(),
+      completed: false,
+    },
+  );
+}
+
+function toggleTodo(
+  state: Array<{
+    uid: string,
+    target: {},
+    actionId: string,
+    completed: boolean}> = [],
+  actionId: string,
+  phoneNumber: {}) {
+  return state.map((todo) => {
+    if (todo.actionId === actionId && todo.target === phoneNumber) {
+      return Object.assign({}, todo, {
+        completed: !todo.completed,
+      });
+    }
+    return todo;
+  });
+}
+
 function todos(
   state: Array<{
     uid: string,
     target: {},
     actionId: string,
     completed: boolean}> = [],
   action: {
-    type: string, actionId?: string, uid?: string, phoneNumbers?: Array<{}>,
+    type: string,
+    actionId: string, uid?: string,
+    phoneNumbers?: Array<{}>,
+    phoneNumber?: {},
   }) {
   switch (action.type) {
+    case ADD_OR_TOGGLE_TODO:
+      return toggleTodo(addTodo(state, action.actionId, action.phoneNumber),
```

Comment:
So, flow is yelling at me here (and several other similar places) because I marked `action.phoneNumber` as optional but don't have any code here to handle the null case.  `addTodo(state, action.actionId, action.phoneNumber || {})` would quiet flow but feels very wrong to me, since `action.phoneNumber` *is* required when `action.type === ADD_OR_TOGGLE_TODO`.

Is there some way to add an inline flow assertion in this code branch to that effect..?

Or alternatively, is there any way to actually specify the relationship between `action.type` and the remainder of the `action` object in the function annotation..?